tests/integration/backward_compatible/sql_test.py

```diff
@@ -0,0 +1,590 @@
+        iterator = result.iterator().result()
+
+        rows = []
+        for row_future in iterator:
+            try:
+                row = row_future.result()
+                rows.append(row.get_object("this"))
+            except StopIteration:
+                break
+
+        six.assertCountEqual(self, [i for i in range(10)], rows)
+
+        with self.assertRaises(ValueError):
+            result.iterator().result()
+
+    def test_iterator_with_multi_paged_result(self):
+        self._populate_map()
+        statement = SqlStatement("SELECT __key FROM %s" % self.map_name)
+        statement.cursor_buffer_size = 1  # Each page will contain just 1 result
+        result = self.client.sql.execute_statement(statement)
+
+        iterator = result.iterator().result()
+
+        rows = []
+        for row_future in iterator:
+            try:
+                row = row_future.result()
+                rows.append(row.get_object_with_index(0))
+            except StopIteration:
+                break
+
+        six.assertCountEqual(self, [i for i in range(10)], rows)
+
+    def test_request_blocking_iterator_after_iterator(self):
+        self._populate_map()
+        result = self.client.sql.execute("SELECT * FROM %s" % self.map_name)
+
+        result.iterator().result()
+
+        with self.assertRaises(ValueError):
+            for _ in result:
+                pass
+
+    def test_request_iterator_after_blocking_iterator(self):
+        self._populate_map()
+        result = self.client.sql.execute("SELECT * FROM %s" % self.map_name)
+
+        for _ in result:
+            pass
+
+        with self.assertRaises(ValueError):
+            result.iterator().result()
+
+    # Can't test the case we would expect row to be not set, because the IMDG SQL
+    # engine does not support update/insert queries now.
+    def test_is_row_set_when_row_is_set(self):
+        self._populate_map()
+        result = self.client.sql.execute("SELECT * FROM %s" % self.map_name)
+        self.assertTrue(result.is_row_set().result())
+
+    # Can't test the case we would expect a non-negative updated count, because the IMDG SQL
+    # engine does not support update/insert queries now.
+    def test_update_count_when_there_is_no_update(self):
+        self._populate_map()
+        result = self.client.sql.execute("SELECT * FROM %s WHERE __key > 5" % self.map_name)
+        self.assertEqual(-1, result.update_count().result())
+
+    def test_get_row_metadata(self):
+        self._populate_map(value_factory=str)
+        result = self.client.sql.execute("SELECT * FROM %s" % self.map_name)
+        row_metadata = result.get_row_metadata().result()
+        self.assertEqual(2, row_metadata.column_count)
+        columns = row_metadata.columns
+        self.assertEqual(SqlColumnType.INTEGER, columns[0].type)
+        self.assertEqual(SqlColumnType.VARCHAR, columns[1].type)
+        self.assertTrue(columns[0].nullable)
+        self.assertTrue(columns[1].nullable)
```

Comment:
```suggestion
    def test_get_row_metadata(self):
        self._populate_map(value_factory=str)
        result = self.client.sql.execute("SELECT * FROM %s" % self.map_name)
        row_metadata = result.get_row_metadata().result()
        self.assertEqual(2, row_metadata.column_count)

        key_column_index = row_metadata.find_column("__key")
        this_column_index = row_metadata.find_column("this")

        key_column = row_metadata.get_column(key_column_index)
        this_column = row_metadata.get_column(this_column_index)

        self.assertEqual(SqlColumnType.INTEGER, key_column.type)
        self.assertEqual(SqlColumnType.VARCHAR, this_column.type)
        self.assertTrue(key_column.nullable)
        self.assertTrue(this_column.nullable)
```

to test public API and to be sure of column indexes

Reply:
changed the query to be sure of column indexes instead of changing the test as you suggested, again due to code coverage. I test something I don't use in other tests here

Reply:
Did the same thing to other places to be sure of column indexes